### ISSUE TITLE
How to align 2K2K scans with SMPL-X?

### ISSUE THREAD

**Issue by azuxmioy** [SangHunHan92/2K2K#13]
Hello,

Thanks for updating the dataset. I've tried the latest SMPL-X parameters in the [dataset repo](https://github.com/ketiVision/2K2K#data-explanation).

I used the following code to create the body mesh

```
body_model = SMPLX(model_path=os.path.join(MODEL_PATH,'smplx'), use_face_contour=True, use_pca=False, gender='neutral')
smpl_data = json.load(open(os.path.join(smplx_folder, smpl_file)))
output = body_model(  global_orient= torch.tensor(smpl_data['global_orient']).unsqueeze(0).contiguous(),
                          transl = torch.tensor(smpl_data['transl']).unsqueeze(0).contiguous(),
                          body_pose = torch.tensor(smpl_data['body_pose']).unsqueeze(0).contiguous(),
                          betas = torch.tensor(smpl_data['betas']).unsqueeze(0).contiguous(),
                          jaw_pose=torch.tensor(smpl_data['jaw_pose']).unsqueeze(0).contiguous(),
                          left_hand_pose = torch.tensor(smpl_data['left_hand_pose']).unsqueeze(0).contiguous(),
                          right_hand_pose = torch.tensor(smpl_data['right_hand_pose']).unsqueeze(0).contiguous(),
                          expression=torch.tensor(smpl_data['expression']).unsqueeze(0).contiguous(),
                          return_verts=True
                        )
```

However, there are still translation and rotation offsets between the PLY scans and the SMPLX body mesh.

![Screen Shot 2023-09-28 at 11 36 38 PM](https://github.com/SangHunHan92/2K2K/assets/18035210/4210fc37-a541-4c80-9517-734eac0b6f47)

Also, it seems that the scale (0.986) is not correct. Could you please describe how you registered the body model parameters?

Thank you very much.

**Comment by fishfishson** [comment]
@azuxmioy Hi, I have met with the same problem. Have you found the solution to this misalignment?

**Comment by fengq1a0** [comment]
@fishfishson Hi, I also have the same problem. Have you found any solution?
(the given SMPL mesh in the .obj file is 170m tall, however, the mesh in the .ply file is only about 1m)

**Comment by fishfishson** [comment]
I have no idea. It seems like scans are rescaled for privacy protection.

________________________________
发件人: fengq1a0 ***@***.***>
发送时间: 2024年4月24日 14:15
收件人: SangHunHan92/2K2K ***@***.***>
抄送: YU Zhiyuan ***@***.***>; Mention ***@***.***>
主题: Re: [SangHunHan92/2K2K] How to align 2K2K scans with SMPL-X? (Issue #13)


@fishfishson<https://github.com/fishfishson> Hi, I also have the same problem. Have you found any solution?
(the given SMPL mesh in the .obj file is 170m tall, however, the mesh in the .ply file is only about 1m)

―

**Comment by yvette256** [comment]
Hi, super late to this thread but I'm currently running into a similar issue. I was wondering if anyone managed to figure out the proper alignment?

To the authors, I also wanted to ask if it's possible to clarify whether the rescaling is indeed intentionally for privacy protection, thank you in advance!

**Comment by azuxmioy** [comment]
Hi all,

This is just an unhelpful update after a year. Since the scale and offset information are missing in the dataset, I ended up registering SMPL-X parameters myself by rendering multi-view images from the scan. But still, the absolute measurement is unknown.

I feel the rescaling is not for privacy protection but a normalization in 3D instead as scans of kids are generally larger. However, the scaling factor and normalization function are not provided, so we don't know the exact measurement of the original human bodies. I'll keep this issue open and see if the authors have any updates.
